### DISCUSSION TITLE
Developer Guides

### DISCUSSION THREAD

**Discussion by scottilee** [pytest-dev/pytest#7889] [Q&A] [answered]
Are there any docs that guide developers through the code?

For instance, why is the code structured under the [_pytest](https://github.com/pytest-dev/pytest/tree/master/src/_pytest) directory?

**Comment by The-Compiler** [comment, marked as answer]
There are the [contribution docs](https://github.com/pytest-dev/pytest/blob/master/CONTRIBUTING.rst) and the [API reference](https://docs.pytest.org/en/latest/reference.html) (many things in there, especially plugin hooks, are relevant for the core as well).

The `_pytest` directory is a Python package with the leading underscore marking it as "internal", while anything that's public gets exposed in the `pytest` package. That's not something specific to pytest though, see e.g. https://stackoverflow.com/a/55178477/2085149 as well.
